view/frontend/web/js/product/button.js

```diff
@@ -0,0 +1,52 @@
+define([
+    'jquery',
+    'underscore'
+],function ($, _) {
+
+    return function (config) {
+
+        $(config.buttonId).on('mouseover',function (e) {
+
+            var radioAndCheckbox = {};
+            var inputElements = [];
+
+            $.each($('.product-options-wrapper [aria-required="true"]'),function () {
+
+                if(typeof $(this).prop("name") === "undefined" || $(this).attr("type") === "hidden"){
+                    return;
+                }
+
+                if($(this).attr("type") === "radio" || $(this).attr("type") === "checkbox"){
+
+                    String($(this).attr("name")) in radioAndCheckbox ?
```

Comment:
could you break this line out into some variables as well? It's pretty long and there's a lot going on that isn't super clear from an initial glance

Reply:
What line, String(... or if condition?